pkg/yttlibrary/overlay/array_item_match_annotation.go

```diff
@@ -18,21 +18,28 @@ type ArrayItemMatchAnnotation struct {
 	newItem *yamlmeta.ArrayItem
 	thread  *starlark.Thread
 
-	matcher *starlark.Value
-	expects MatchAnnotationExpectsKwarg
+	matcher     *starlark.Value
+	expects     MatchAnnotationExpectsKwarg
+	unannotated bool
 }
 
 func NewArrayItemMatchAnnotation(newItem *yamlmeta.ArrayItem,
 	defaults MatchChildDefaultsAnnotation,
 	thread *starlark.Thread) (ArrayItemMatchAnnotation, error) {
 
 	annotation := ArrayItemMatchAnnotation{
-		newItem: newItem,
-		thread:  thread,
-		expects: MatchAnnotationExpectsKwarg{thread: thread},
+		newItem:     newItem,
+		thread:      thread,
+		expects:     MatchAnnotationExpectsKwarg{thread: thread},
+		unannotated: false,
 	}
 	anns := template.NewAnnotations(newItem)
 
+	if len(anns) == 0 {
```

Comment:
this is a bit problematic. what if my node has #@yaml/text-templated-strings or any other annotation. i think we need to push this functionality in the if conditional below.